### PR TITLE
DB load history

### PR DESCRIPTION
**Pull request by shreyas-damle**
- Implemented load history.
- Handled "Findings" on "Documents With Findings" tab on application details page.  For this, one change needed on the `aidocument` table about adding findings count per entity/topic.
- Pydantic fixes needed due to upgrade to pydantic v2 => `dict()` to `model_dump()`.
- timeit() would run only when debug logs are enabled.
- Delete pdf report when delete app is called from UI.

### CODE REVIEW

File: pebblo/app/service/loader/document/document.py [hunk 1]
```diff
@@ -102,17 +102,23 @@ def _update_document(document, snippet):
...
-                    existing_entities.update({entity: {"ref": [snippet.get("id")]}})
+                    existing_entities.update(
+                        {entity: {"ref": [snippet.get("id")], "count": 1}}
+                    )
         if topics:
             for topic in topics:
                 if topic in existing_topics.keys():
                     updated_topic = existing_topics[topic]
                     updated_topic["ref"].append(snippet.get("id"))
+                    updated_topic["count"] = updated_topic.get("count") + 1
```
Comment (gr8nishan) [inline pebblo/app/service/loader/document/document.py:116]: We can use updated_topic["count"]  += 1 for better readability

File: pebblo/app/service/loader/document/document.py [hunk 2]
```diff
@@ -123,7 +129,6 @@ def _update_document(document, snippet):
     def create_or_update_document(self, app_loader_details, data_source):
```
Comment (gr8nishan) [inline pebblo/app/service/loader/document/document.py:129]: Can we add type in function params and output type

Comment (gr8nishan) [reply pebblo/app/service/loader/document/document.py:129]: Also comments will be helpful

File: pebblo/app/service/local_ui/loader_apps.py [hunk 3]
```diff
@@ -28,15 +31,24 @@
...
+            "loader_findings_list": [],
+            "loader_findings": 0,
+            "loader_data_source_list": [],
+            "loader_data_source_count": 0,
+            "loader_document_with_findings_list": [],
+            "loader_files_with_findings_count": 0,
+            "loader_findings_summary_list": [],
+        }
+
     def _get_snippet_details(self, snippet_ids: list, owner: str, label_name: str):
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:52]: Can we add type for function output

File: pebblo/app/service/local_ui/loader_apps.py [hunk 4]
```diff
@@ -287,6 +314,23 @@ def get_findings_for_loader_app(self, app_data):
         )
         return app_details.model_dump()
 
+    def _create_loader_app_model(self, app_list):
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:317]: Lets add some comments and param and output type

File: pebblo/app/service/local_ui/loader_apps.py [hunk 5]
```diff
@@ -346,28 +383,32 @@ def get_all_loader_apps(self):
             self.db.session.close()
 
     def get_loader_app_details(self, db, app_name):
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:385]: Lets add function param and output type

File: pebblo/app/service/local_ui/loader_apps.py [hunk 6]
```diff
@@ -379,11 +420,11 @@ def get_loader_app_details(self, db, app_name):
             logger.debug(message)
             return json.dumps(report_data, default=str, indent=4)
 
-    def _create_report_summary(self, raw_data, app_data):
+    @staticmethod
+    def _create_report_summary(raw_data, app_data):
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:424]: Lets add function param type and output type

File: pebblo/app/service/loader/document/document.py (hunk 2 above)
Comment (shreyas-damle) [reply pebblo/app/service/loader/document/document.py:129]: Fixed.

File: pebblo/app/service/local_ui/loader_apps.py (hunk 3 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py:52]: Fixed.

File: pebblo/app/service/local_ui/loader_apps.py (hunk 4 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py:317]: Fixed.

File: pebblo/app/service/local_ui/loader_apps.py (hunk 5 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py:385]: Added.

File: pebblo/app/service/local_ui/loader_apps.py (hunk 6 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py:424]: Added.

File: pebblo/app/service/local_ui/loader_apps.py [hunk 7]
```diff
@@ -345,29 +406,33 @@ def get_all_loader_apps(self):
             # Closing the session
             self.db.session.close()
 
-    def get_loader_app_details(self, db, app_name):
+    def get_loader_app_details(self, db: SQLiteClient, app_name: str):
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:409]: return type is missing

File: pebblo/app/service/local_ui/loader_apps.py [hunk 8]
```diff
@@ -398,7 +463,7 @@ def _create_report_summary(self, raw_data, app_data):
         return report_summary
 
     @staticmethod
-    def _get_top_n_findings(raw_data):
+    def _get_top_n_findings(raw_data: dict):
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:466]: return type is missing

File: pebblo/app/service/local_ui/loader_apps.py [hunk 9]
```diff
@@ -425,7 +490,77 @@ def _get_top_n_findings(raw_data):
...
+                    loadId=load_id,
+                    reportName=report_name,
+                    findings=report_summary["findings"],
+                    filesWithFindings=report_summary["filesWithFindings"],
+                    generatedOn=report_summary["createdAt"],
+                )
+                load_history["history"].append(load_history_model_obj.model_dump())
+                load_history_instances += 1
+            except Exception:
+                logger.error(
```
Comment (gr8nishan) [inline pebblo/app/service/local_ui/loader_apps.py:548]: Should this be warning ?

File: pebblo/app/service/local_ui/loader_apps.py (hunk 7 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py]: Fixed.

File: pebblo/app/service/local_ui/loader_apps.py (hunk 8 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py]: fixed.

File: pebblo/app/service/local_ui/loader_apps.py (hunk 9 above)
Comment (shreyas-damle) [reply pebblo/app/service/local_ui/loader_apps.py:546]: Fixed.

**Review by gr8nishan — DISMISSED** [review]

**Review by gr8nishan — APPROVED** [review]

**Pull request merged**